python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint.py

```diff
@@ -321,76 +428,241 @@ def toString(self, engine="Gromacs"):
 
         engine : str
             The molecular dynamics engine used to generate the restraint.
-            Available options currently is "GROMACS" only. If this argument
+            Available options currently is "GROMACS" and "SOMD". If this argument
             is omitted then BioSimSpace will choose an appropriate engine
             for you.
         """
         if engine.strip().lower() == "gromacs":
             if self._restraint_type == "boresch":
                 return self._gromacs_boresch()
+        elif engine.lower() == "somd":
+            if self._restraint_type == "boresch":
+                return self._somd_boresch()
             else:
                 raise NotImplementedError(
-                    f"Restraint type {self.restraint_type} not implemented "
-                    f"yet. Only boresch restraint is supported."
+                    f"Restraint type {self._restraint_type} not implemented "
+                    f"yet. Only Boresch restraints are supported."
                 )
         else:
             raise NotImplementedError(
                 f"MD Engine {engine} not implemented "
-                f"yet. Only Gromacs is supported."
+                f"yet. Only Gromacs and SOMD are supported."
             )
 
-    @property
-    def correction(self):
-        """Give the free energy of removing the restraint."""
-        if self._restraint_type == "boresch":
-            K = _k_boltz.value() * (
-                _kcal_per_mol / _kj_per_mol
-            )  # Gas constant in kJ/mol/K
-            V = (
-                (_meter3 / 1000 / _mole) / _nanometer3
-            ).value()  # standard volume in nm^3 (liter/N_A)
+    def getCorrection(self, method="numerical"):
```

Comment:
I changed the default to numerical to avoid any errors introduced by the analytical formula and the time penalty is negligible. Why would you prefer the analytical method? However, I've only ever seen substantial differences between the the methods when the force constants are very weak or the restraints are unstable, so happy to change the default if you'd much prefer analytical.

Reply:
I think it is more like numerical integration error has hurt me too many times that I prefer to do analytical method when it is available. I think people usually go for the analytical method when they could and only switch to numerical when analytical one is not available?

Reply:
No problem, I'll change the default to analytical. 

Just to note that I noticed this on [the Yank documentation](http://getyank.org/latest/algorithms.html#restraints-and-standard-state-correction):
>The standard state correction is computed by evaluating using a combination of numerical and analytical one-dimensional integrals from Eq. 12 of [[18]](http://getyank.org/latest/references.html#boresch2003). Note that the analytical standard state correction described in Eq. 32 of [[18]](http://getyank.org/latest/references.html#boresch2003) is inaccurate (up to several kT) in certain regimes (near raA0 and θA0, θB0 near 0 or π) and should be avoided.

>Due to numerical instabilities and the generalization of the Boresch restraints, the analytical free energy contribution is no longer supported.

Reply:
I know that analytical correction is inaccurate when angles approaching the 0 or π. But I'm a bit surprised that numerical solution can deal with this (Without explicitly account for this)?